source/common/http/http2/codec_impl.cc

```diff
@@ -1670,7 +1687,8 @@ ConnectionImpl::Http2Callbacks::Http2Callbacks() {
 
   nghttp2_session_callbacks_set_on_begin_headers_callback(
       callbacks_, [](nghttp2_session*, const nghttp2_frame* frame, void* user_data) -> int {
-        auto status = static_cast<ConnectionImpl*>(user_data)->onBeginHeaders(frame);
+        Status status =
+            static_cast<ConnectionImpl*>(user_data)->onBeginHeaders(frame->hd.stream_id);
```

Comment:
@birenroy im not sure if i resolved the conflict here correctly

its erroring - my plan was to let it error to find the affected code paths - but i think its erroring more generally with this patch